pipeline/stage02_processing/scripts/spatial_downsampling.py

```diff
@@ -31,45 +51,24 @@ def spatial_smoothing(images, MACRO_PIXEL_DIM):
                      help="path of output file")
     CLI.add_argument("--output_img",  nargs='?', type=none_or_str,
                      help="path of output image", default=None)
-    CLI.add_argument("--output_array",  nargs='?', type=none_or_str,
-                      help="path of output numpy array", default=None)
 
     CLI.add_argument("--macro_pixel_dim",  nargs='?', type=int,
-                      help="smoothing factor", default=2)
+                      help="smoothing factor", default=4)
 
     
     args = CLI.parse_args()
 
     block = load_neo(args.data)
     block = AnalogSignal2ImageSequence(block)
-    imgseq = block.segments[0].imagesequences[-1]
+    imgseq = block.segments[0].imagesequences[0]
     
-    MACRO_PIXEL_DIM = args.macro_pixel_dim    
-    img_reduced = spatial_smoothing(imgseq, MACRO_PIXEL_DIM)
-
-    if args.output_array is not None:
-        np.save(args.output_array, img_reduced[0])
-    if args.output_img is not None:
-        plt.figure()
-        plt.imshow(img_reduced[0])
-        save_plot(args.output_img)
-
-    dim_t = len(img_reduced[:,0,0])
-    dim_x = len(img_reduced[0,:,0])
-    dim_y = len(img_reduced[0,0,:])
-
-    imgseq_reduced = neo.ImageSequence(img_reduced,
-                                   units='dimensionless',
-                                   spatial_scale=imgseq.spatial_scale * MACRO_PIXEL_DIM,
-                                   sampling_rate=imgseq.sampling_rate,
-                                   name='Reduced Images',
-                                   description='Spatial downsampling',
-                                   file_origin=imgseq.file_origin,
-                                   **imgseq.annotations)
+    MACRO_PIXEL_DIM = args.macro_pixel_dim
+    print('Macro', MACRO_PIXEL_DIM)
+    imgseq_reduced = spatial_smoothing(imgseq, MACRO_PIXEL_DIM)
```

Comment:
To address some minor details, in the pipeline we try to use the variable naming convention that only global variables are capitalized (i.e. config parameters in the Snakefile), otherwise, the _snake_case_ naming should be used. Besides, it is here unnecessary to rename args.macro_pixel_dim which is anyway only used once immediately after.